src/shared/utils/deprecated.js

```diff
@@ -18,30 +18,26 @@ var warning = require('warning');
  * This will log a single deprecation notice per function and forward the call
  * on to the new API.
  *
- * @param {string} namespace The namespace of the call, eg 'React'
- * @param {string} oldName The old function name, eg 'renderComponent'
- * @param {string} newName The new function name, eg 'render'
+ * @param {string} fnName The name of the function
+ * @param {string} newModule The module that fn will exist in
  * @param {*} ctx The context this forwarded call should run in
  * @param {function} fn The function to forward on to
- * @return {*} Will be the value as returned from `fn`
+ * @return {function} The function that will warn once and then call fn
  */
-function deprecated(namespace, oldName, newName, ctx, fn) {
+function deprecated(fnName, newModule, ctx, fn) {
   var warned = false;
   if (__DEV__) {
     var newFn = function() {
       warning(
         warned,
-        '%s.%s will be deprecated in a future version. ' +
-        'Use %s.%s instead.',
-        namespace,
-        oldName,
-        namespace,
-        newName
+        '`React.%s` will be deprecated from the `react` package in a future ' +
+        'version. It will be available in the `%s` package instead.',
```

Comment:
How about some that shows the from->to transformation directly? Easier to understand "you did __this__ but you should be doing __that__".

```js
'require('react').%s` is deprecated, use `require('%s').%s` instead?'
```